tools/publish-aos.js

```diff
@@ -1,54 +1,170 @@
+    process.exit(0); // Exit with code 0 to indicate success
+  }
+  let wallet = process.env.ARWEAVE_PUBLISHING_KEY;
+  if (!wallet) {
+    try {
+      wallet = fs.readFileSync(path.join(__dirname, 'key.json'), 'utf-8');
+    } catch (error) {
+      console.error(
+        'No key found, please set ARWEAVE_PUBLISHING_KEY environment variable or provide a key.json file',
+      );
+      return;
+    }
+  }
   const jwk = JSON.parse(wallet);
   const address = await arweave.wallets.jwkToAddress(jwk);
+  const signer = new ArweaveSigner(jwk);
+  const turbo = TurboFactory.authenticated({ signer });
 
-  const tx1 = await arweave.createTransaction({ data: bundledLua }, jwk);
-  tx1.addTag('App-Name', 'aos-LUA');
-  tx1.addTag('App-Version', '0.0.1');
-  tx1.addTag('Content-Type', 'text/x-lua');
-  tx1.addTag('Author', 'Permanent Data Solutions');
-  await arweave.transactions.sign(tx1, jwk);
-  await arweave.transactions.post(tx1);
+  const publishingTags = Object.entries({
+    'App-Name': 'aos-LUA',
+    'App-Version': '0.0.1',
+    'Content-Type': 'text/x-lua',
+    Author: 'Permanent Data Solutions',
+    Changelog: changelog,
+  }).map(([name, value]) => ({ name, value }));
 
-  // replace placeholder with actual tx id
-  const bundledLuaWithTxId = bundledLua.replace(srcCodeTxIdPlaceholder, tx1.id);
+  const data1 = await createData(bundledLua, signer, {
+    tags: publishingTags,
+  });
+  await data1.sign(signer);
+  const dataTx1 = await turbo.uploadSignedDataItem({
+    dataItemSizeFactory: () => data1.getRaw().byteLength,
+    dataItemStreamFactory: () => data1.getRaw(),
+  });
 
-  console.log(`Publish AOS ANT Lua with address ${address}`);
+  // replace placeholder with actual tx id
+  const bundledLuaWithTxId = bundledLua.replace(
+    srcCodeTxIdPlaceholder,
+    dataTx1.id,
+  );
 
-  const tx = await arweave.createTransaction({ data: bundledLuaWithTxId }, jwk);
-  tx.addTag('App-Name', 'aos-LUA');
-  tx.addTag('App-Version', '0.0.1');
-  tx.addTag('Content-Type', 'text/x-lua');
-  tx.addTag('Author', 'Permanent Data Solutions');
-  tx.addTag('Original-Tx-Id', tx1.id);
-  tx.addTag(
-    'Changelog',
-    `# Changelog
+  const data2 = await createData(bundledLuaWithTxId, signer, {
+    tags: [...publishingTags, { name: 'Original-Tx-Id', value: dataTx1.id }],
+  });
+  await data2.sign(signer);
 
-### Changed
+  const dataTx2 = await turbo.uploadSignedDataItem({
+    dataItemSizeFactory: () => data2.getRaw().byteLength,
+    dataItemStreamFactory: () => data2.getRaw(),
+  });
 
-- Changed default landing page transaction ID.
-`,
+  // create changelog and write to changelogs folder
+  const currentDate = new Date();
+  const year = currentDate.getFullYear();
+  const month = String(currentDate.getMonth() + 1).padStart(2, '0');
+  const day = String(currentDate.getDate()).padStart(2, '0');
+  const formattedDate = `${year}-${month}-${day}`;
+  const changelogPath = path.join(
+    __dirname,
+    '../changelogs',
```

Comment:
oy - is this necessary for arns.app? i'd avoid this changelog approach. so long as we make note of in github that should be enough